src/queryMetrics/clientSideMetrics.ts

```diff
@@ -0,0 +1,58 @@
+import { QueryMetricsUtils } from "./queryMetricsUtils";
+
+export class ClientSideMetrics {
+    private requestCharge: number;
+    constructor(requestCharge: number) {
+        // Constructor
+        if (!QueryMetricsUtils.isNumeric(requestCharge)) {
+            throw new Error("requestCharge is not a numeric type");
+        }
+
+        this.requestCharge = requestCharge;
+    }
+
+    /**
+     * Gets the RequestCharge
+     * @memberof ClientSideMetrics
+     * @instance
+     * @ignore
+     */
+    public getRequestCharge() {
+        return this.requestCharge;
+    }
+
+    /**
+     * Adds one or more ClientSideMetrics to a copy of this instance and returns the result.
+     * @memberof ClientSideMetrics
+     * @instance
+     */
+    public add(clientSideMetricsArray: ClientSideMetrics[]) {
+        if (arguments == null || arguments.length === 0) {
+            throw new Error("arguments was null or empty");
+        }
+
+        clientSideMetricsArray.push(this);
```

Comment:
push to a copy of clientSideMetricsArray so we don't modify the input.

Reply:
That's wasteful. Instead, I'll initialize requestCharge to this.requestCharge.